google/cloud/storage/internal/curl_download_request.cc

```diff
@@ -48,11 +48,14 @@ HttpResponse CurlDownloadRequest::Close() {
   auto error = curl_multi_remove_handle(multi_.get(), handle_.handle_.get());
   RaiseOnError(__func__, error);
```

Comment:
Are the RaiseOnError() calls in StatusOr-returning functions considered "cannot happen", or just fodder for another day?